sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/trees/TreeNode.scala

```diff
@@ -397,6 +393,7 @@ abstract class TreeNode[BaseType <: TreeNode[BaseType]] extends Product {
   def argString: String = productIterator.flatMap {
     case tn: TreeNode[_] if containsChild(tn) => Nil
     case tn: TreeNode[_] => s"${tn.simpleString}" :: Nil
+    case seq: Seq[_] if seq == Seq.empty => "[Empty]" :: Nil
```

Comment:
`if seq.isEmpty`.

That said, it seems a little confusing to have it be `[Empty]` and not just `[]`.  Is there a reason the case below isn't sufficient?